core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/OperationChainLimiter.java

```diff
@@ -165,16 +164,24 @@ public int compare(final Entry<K, V> entry1, final Entry<K, V> entry2) {
     @Override
     public void preExecute(final OperationChain<?> opChain, final User user) {
         if (null != opChain) {
-            Integer chainScore = 0;
+            Integer chainScore = getChainScore(opChain, user);
             Integer maxAuthScore = getMaxUserAuthScore(user.getOpAuths());
+            if (chainScore > maxAuthScore) {
+                throw new UnauthorisedException("The maximum score limit for this user is " + maxAuthScore + ".\n" +
+                        "The requested operation chain exceeded this score limit.");
+            }
+        }
+    }
+
+    public Integer getChainScore(final OperationChain<?> opChain, final User user) {
+        Integer chainScore = 0;
```

Comment:
Should we be adopting a standard practice of using primitives everywhere unless the boxed version is specifically required?